### PR TITLE
Updated IPC tests

### PR DESCRIPTION
**Pull request by RusherRG**
## Description

 

Updated the IPC-performance tests to include user tests using [native-rustposix](https://github.com/Lind-Project/native-rustposix). I also updated the plotting scripts to include the error for each bar.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

- total_runtime - `lind_project/tests/ipc_performance/total_runtime`

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] Any dependent changes have been added to a pull request and/or merged in other modules (native-client, lind-glibc, safeposix-rust)

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -21,3 +21,9 @@
 [submodule "src/safeposix-rust"]
 	path = src/safeposix-rust
 	url = https://github.com/Lind-Project/safeposix-rust.git
+[submodule "tests/ipc_performance/total_runtime/scripts/native-rustposix"]
+	path = tests/ipc_performance/total_runtime/scripts/native-rustposix
+	url = https://github.com/Lind-Project/native-rustposix.git
+[submodule "tests/ipc_performance/read_write_vcs/scripts/native-rustposix"]
```
Comment (rennergade) [inline .gitmodules:27]: is this supposed to be added twice?

Comment (RusherRG) [reply .gitmodules:27]: Yeah I think because it is for different tests.

Or I could just add this to the `tests/` folder

Comment (RusherRG) [reply .gitmodules]: Addressed this by moving the `native-rustposix` submodule to `tests/`

**Review by rennergade — APPROVED** [review]
Awesome!

**Pull request merged**